tools/scxtop/src/app.rs

```diff
@@ -348,8 +370,51 @@ impl<'a> App<'a> {
         self.max_cpu_events = max_events;
     }
 
+    /// Handles when scheduler stats are received.
+    fn on_sched_stats(&mut self, stats_raw: String) {
+        self.sched_stats_raw = stats_raw;
+    }
+
     /// Runs callbacks to update application state on tick.
     fn on_tick(&mut self) -> Result<()> {
+        match self.state {
+            AppState::Scheduler => {
+                if !self.scheduler.is_empty() {
+                    let stats_client_read = self.stats_client.clone();
+                    let stats_reconnect = self.stats_reconnect.clone();
+
+                    let tx = self.action_tx.clone();
+                    tokio::spawn(async move {
+                        let mut client = stats_client_read.write().await;
+                        if stats_reconnect.load(Ordering::Relaxed) {
```

Comment:
I think if this gets moved into the error handling the bool isn't needed now that I think about it more.